lib/hsa/hsaBeamformPulsar.cpp

```diff
@@ -23,29 +22,18 @@ hsaBeamformPulsar::hsaBeamformPulsar(Config& config, const std::string& unique_n
     command_type = gpuCommandType::KERNEL;
 
     _num_elements = config.get<int32_t>(unique_name, "num_elements");
-    _num_pulsar = config.get<int32_t>(unique_name, "num_beams");
+    _num_beams = config.get<int32_t>(unique_name, "num_beams");
```

Comment:
Can we add a check that `_num_beams <= MAX_NUM_BEAMS` and throw a runtime exception if not?